File: Data Structures & Algorithms/buy-and-sell-crypto/submission-2.cs
public class Solution {
    public int MaxProfit(int[] prices) {
        var l = 0;
        var max = 0;
        for(var r = 1; r < prices.Length; r++)
        {
            if(prices[l] < prices[r])
            {
                max = Math.Max(max, prices[r] - prices[l]);
            }else{
                l = r;
            }
        }
        return max;
    }
}

File: Data Structures & Algorithms/jump-game/submission-2.py
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        length = len(nums)
        maxReach = 0
        for i in range(length):
            if i > maxReach : return False
            maxReach = max(maxReach, i + nums[i])
        return True
